CMakeLists.txt

```diff
@@ -748,9 +748,23 @@ if(IREE_BUILD_DOCS)
   add_custom_target(iree-doc ALL)
 endif()
 
+# Samples may require additional files to be built/configured and will add
+# dependencies to this target.
+# Note: These will be automatically built with test dependencies
+# (`iree-test-deps`).
+add_custom_target(iree-sample-deps
+  COMMENT
+    "Building IREE sample data targets"
+)
+
 # Testing rules that require generation will add dependencies to this target.
 # This allows them to be EXCLUDE_FROM_ALL but still invokable.
-add_custom_target(iree-test-deps COMMENT "Building IREE test deps")
+add_custom_target(iree-test-deps
+  COMMENT
+    "Building IREE test deps"
+  DEPENDS
+    iree-sample-deps
```

Comment:
Ah, this is cleaner than what https://github.com/iree-org/iree/pull/11287 had originally
![image](https://user-images.githubusercontent.com/4010439/217362630-a35929a0-2fd2-4065-802f-093d5762177a.png)
Nice.